src/utils/io/progress_log.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
```

Comment:
I think this needs a module comment, explaining
- what this is / how it behaves
- _why_ it exists
- when it should be used - _long_ running operations, like reading a large file
- usage principles - this _wraps_ APIs, and you shouldn't be making 'inner' calls